docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1345,444 +1370,119 @@ To create, accept, or cancel escrows, it's better to implement mutations for eac
 
+{@inject: examples/trading/frontend/src/components/locked/LockedObject.tsx}
+
 The `<Locked />` (`src/components/locked/partials/Locked.tsx`) component is mainly responsible for rendering the `ApiLockedObject`. It also consists of several on-chain interactions: unlock the locked objects and create an escrow out of the locked object.
 
-```ts title='src/components/locked/LockedObject.tsx'
-/**
- * Acts as a wrapper between the `Locked` object fetched from API
- * and the on-chain object state.
- *
- * Accepts an `object` of type `::locked::Locked`, fetches the itemID (though the DOF)
- * and then renders the `Locked` component.
- *
- * ItemId is optional because we trust the API to return the correct itemId for each Locked.
- */
-export function LockedObject({
-	object,
-	itemId,
-	hideControls,
-}: {
-	object: SuiObjectData;
-	itemId?: string;
-	hideControls?: boolean;
-}) {
-	const owner = () => {
-		if (!object.owner || typeof object.owner === 'string' || !('AddressOwner' in object.owner))
-			return undefined;
-		return object.owner.AddressOwner;
-	};
+{@inject: examples/trading/frontend/src/components/locked/partials/Locked.tsx}
 
-	const getKeyId = (item: SuiObjectData) => {
-		if (!(item.content?.dataType === 'moveObject') || !('key' in item.content.fields)) return '';
-		return item.content.fields.key as string;
-	};
+The `Locked` component is a bit complex because it has multiple actions that can be performed on the locked object. The `Locked` component is responsible for rendering the `ApiLockedObject` and providing the necessary actions to interact with the locked object. The `Locked` component consists of the following actions:
 
-	// Get the itemID for the locked object (We've saved it as a DOF on the SC).
-	const suiObjectId = useSuiClientQuery(
-		'getDynamicFieldObject',
-		{
-			parentId: object.objectId,
-			name: {
-				type: CONSTANTS.escrowContract.lockedObjectDFKey,
-				value: {
-					dummy_field: false,
-				},
-			},
-		},
-		{
-			select: (data) => data.data,
-			enabled: !itemId,
-		},
-	);
-
-	return (
-		<Locked
-			locked={{
-				itemId: itemId || suiObjectId.data?.objectId!,
-				objectId: object.objectId,
-				keyId: getKeyId(object),
-				creator: owner(),
-				deleted: false,
-			}}
-			hideControls={hideControls}
-		/>
-	);
-}
-```
+- **Unlock** button to unlock the locked object. 
+- **Create Escrow** button to create an escrow out of the locked object. This contains a complex UI so it is implemented in `src/components/escrows/CreateEscrow.tsx`.
+
+{@inject: examples/trading/frontend/src/components/escrows/CreateEscrow.tsx}
```

Comment:
This snippet could do with some explanation of `useSuiClientInfiniteQuery

Reply:
The `useSuiClientInfiniteQuery` hook is explained here: http://localhost:3000/guides/developer/app-examples/trustless-swap#display-owned-locked-objects. I don't remember if this was before or after your linked section before my changes, but now it's before. I think it's best just to keep that once explanation, if that makes sense with you!